server/athenian/api/controllers/goal_controller.py

```diff
@@ -420,6 +425,19 @@ async def _parse_update_request(
     else:
         jira_issue_types = sorted({normalize_issue_type(t) for t in update_req.jira_issue_types})
 
+    # validate the metric parameters in team goals
+    team_goals_validator = _get_update_team_goals_validator(request.app)
+    team_goals_pseudo_instance = {
+        "metric": goal[Goal.metric.name],
+        "team_goals": [tg.to_dict() for tg in update_req.team_goals],
+    }
+    try:
+        team_goals_validator.validate(team_goals_pseudo_instance)
+    except jsonschema.ValidationError as e:
+        raise ResponseError(
+            InvalidRequestError(".team_goals", f'Invalid team goals: "{e.message}"'),
+        )
```

Comment:
To me, this is an indicator that we are misusing OpenAPI. These checks should happen automatically. Are we underdefining the spec?

Look:
- We first deserialize JSON to dicts
- Then validate them over OpenAPI spec
- Then make the request object, validating them further
- Then we convert a part of the request back to dicts
- Then we validate those dicts over another OpenAPI spec

This is the only such place and it is not worth the trouble. We should hardcode all the checks here without feeling shame.

Reply:
The problem here is that the spec cannot express the restriction.
The restriction is about validating a piece of the request (metric_params) with something that it's not in the GoalUpdateRequest, but in the target object on the DB.
By creating a new object with can reuse the existing schema for the GoalCreateRequest

alternatives are:
A. write some procedural code to find the metric type and then manually call the correct subschema for metric_params
B. write completely procedural code for the validation of metric_params against metric
C. decide that  the ~80 lines to do B are not worth the effort and sacrifice metric_params validation for update_goal

Reply:
implemented with procedural code, PTAL